tests/integration/helpers/localize_bundle.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+import copy
+from pathlib import Path
+from typing import Dict, Optional, Union
+import yaml
+from zipfile import ZipFile
+
+
+# TODO: Move this somewhere more general
+
+
+def get_charm_name(metadata_file: Union[Path, str]) -> str:
+    metadata = yaml.safe_load(Path(metadata_file).read_text())
+    return metadata["name"]
+
+
+def get_charm_file(charm_dir: Path) -> Path:
+    """Returns the path to the .charm file representing the charm in the given directory
+
+    TODO: This just assumes the suffix on the file name will be "ubuntu-20.04-amd64".
+          Fix this in future
```

Comment:
Might be a bit overkill and not a long term fix, but you could check the bases in the `charmcraft.yaml` of each charm.

Reply:
Leaving as is for now, but we should do this in future.  Probably this can be pulled from `charmcraft.yaml`, but `arch` is not obviously a property in there.  There's a question outstanding to the starcraft team